### DISCUSSION TITLE
Soundcloud Music provider

### DISCUSSION THREAD

**Discussion by marcelveldt** [music-assistant/support#1160] [Music providers]
MA has support for Soundcloud as a Music provider

Documentation https://music-assistant.io/music-providers/soundcloud/

**Comment by quanghd96** [comment]
Response always is "Session is closed". So I can not add Soundcloud Provider. Please check!

**Reply by marcelveldt** [reply]
this will be fixed in beta 13

**Comment by gitandriu** [comment]
It is only showing the client ID for me, not the o-auth. It is not appended in the URL not show in the headers.  I have tried on Firefox and Edge.

Any tips?

**Reply by OzGav** [reply]
What are you referring to when you use the word “It”?

**Reply by gitandriu** [reply]
I follow the instructions above and the browser's inspect console does not provide o-Auth anywhere, only clientID

**Reply by OzGav** [reply]
Sounds like a browser problem. Try a different browser.

**Reply by sltwin** [reply]
I have the same but could find back Oauth V2 under "tracks" so far so good it works.

**Reply by OzGav** [reply]
A screenshot might help others find it?

**Comment by gitandriu** [comment]
I have tried Edge on Windows and Firefox on a Linux laptop. Same result.

**Reply by OzGav** [reply]
I don’t use edge so can only suggest trying chrome

**Reply by gitandriu** [reply]
I was skeptic since Edge is supposedly based on Chromium, but installed proper Chrome and voilà!, there it was. 

Thanks for the tip, all working great now.

**Reply by steveof2620** [reply]
I am using chrome but I don't see the OAuth anywhere. Have the client_id but not O-Auth.

**Reply by steveof2620** [reply]
Scrub that. I refreshed with page and there it was! Leaving this hear in case it may help others.

**Reply by gieljnssns** [reply]
Just tried this again with arc(chromium based browser) on a mac and I found them again.
The `client id` is in General - Request URL
And the `Authorization` is in Request Headers

**Comment by pejotigrek** [comment]
chrome: 119.0.6045.200
music assistant server [running in docker]: 2.0.0b78

and while configuring SoundCloud with proper client_id and authorisation code it fails with:
`0, message='Attempt to decode JSON with unexpected mimetype: ', url=URL('https://api-v2.soundcloud.com/me')`

**Reply by OzGav** [reply]
Thanks. Has been reported on discord as well. Dev is aware.

**Reply by steveof2620** [reply]
I have the same issue. I take it we use "OAuth" at the start of the Authorisation field.

**Reply by steveof2620** [reply]
Has this issue been fixed yet? Still have this error...
<img width="448" alt="Screenshot 2024-01-22 at 21 21 54" src="https://github.com/music-assistant/hass-music-assistant/assets/46661819/1dd5b3bd-8d36-46d3-b447-0fd7ace6e969">

**Reply by OzGav** [reply]
@gieljnssns any updates?

**Reply by gieljnssns** [reply]
Is your `client_id` like this `TGBnESm7Fol0Ztz2NzLFGrhF0cuKeOb4`?
And your `Authorization`  like this `OAuth 2-294630-4263399-F51Mn3Cr04nfxuVI`

**Reply by steveof2620** [reply]
Yes, I take it I include the OAuth at a the beginning of the string.

**Reply by OzGav** [reply]
yes

**Reply by steveof2620** [reply]
Oh OK, that is what I am putting in.

**Reply by OzGav** [reply]
So is it working?

**Reply by steveof2620** [reply]
No, still have that error. I'll check the credentials again.

**Reply by steveof2620** [reply]
I seem to have a new client ID now, but I don't see the OAuth.

**Reply by steveof2620** [reply]
Found it in another element. Working now. Thanks for your help.

**Reply by pejotigrek** [reply]
where did you found the oauth? I've searched and still no luck...

**Reply by steveof2620** [reply]
It's tricky to find. I used the instructions above but found myself having to refresh the screen. I found it in an his?client_id= string. Had to look through a few of them to find it. I was using chrome.

**Reply by pejotigrek** [reply]
found it!
it was at:
column `name`: session?client_id=_MY_CLIENT_ID_ and I had to search for `access_token` in the tab `payload`

**Comment by steveof2620** [comment]
Just wondering if it would be possible to use an API key in the same way that SqueezeCloud does.

<img width="438" alt="Screenshot 2023-12-30 at 11 53 36" src="https://github.com/music-assistant/hass-music-assistant/assets/46661819/202c882c-7101-4c42-b4cc-b49ce262e477">

**Reply by gieljnssns** [reply]
I don't think this is possible.
This music provider uses an [undocumented, internal version](https://stackoverflow.com/questions/37221510/is-soundcloud-api-v2-being-deprecated) of SoundCloud API.
And I think if we wan't to use the official api, everyone has to register an application on [your account](https://soundcloud.com/signin?redirect_url=/you/apps).
And the problem is, soundcloud isn't planning to [reopen the self-serve app credential creation](https://github.com/soundcloud/api/issues/246#issuecomment-1649629657).

**Comment by Fofer** [comment]
I am new to Music Assistant, and blown away by how great it is, thank you so much!  This Soundcloud integration is working nicely too.  When I am on an artist's page though, the sorting options say:

Alphabetically
Alphabetically
Album

I am wondering if there's any way to have it also sortable by date (newest to oldest and oldest to newest?)

**Reply by OzGav** [reply]
Make sure you are on the latest version and do a browser refresh because you should see this 
![IMG_1249](https://github.com/music-assistant/hass-music-assistant/assets/19848947/bcc835bb-d49f-41ac-9aac-b2427276a2bf)

**Reply by Fofer** [reply]
Thanks, yes, I'm on the latest, now 2.0.2.  When I am in a list of artists, those sorting options are available. But when I select one artist, I only see these:

![IMG_5969](https://github.com/music-assistant/hass-music-assistant/assets/518749/9726d7dd-a6d5-4e57-879f-30656ef8c49c)

So for example when in SoundCloud I want to listen to the most recent track uploaded by my favorite DJ, Lushbunny.  How do I find that track, in Music Assistant?

**Reply by OzGav** [reply]
Seems to be a bug we will look into that

**Reply by Fofer** [reply]
Thank you, very much appreciated. Music Assistant is fantastic!

**Comment by DrDagless** [comment]
I've been setting up Music Assistant over the past week and just added Soundcloud successfully the other day. Also noticed the sorting issue highlighted above, but had another quick question - is the maximum number of tracks 30? 

I use Soundcloud mainly for NTS Radio, since that's where they archive their shows as Tracks. Definitely not ideal, but they've got huge amounts of shows archived there.

Absolutely loving Music Assistant as well!

**Reply by OzGav** [reply]
No the max number of tracks is not 30. Possibly a provider problem. You can open an issue if you would like this investigated

**Comment by Traxtar928** [comment]
I'm getting an error when trying to open SoundCloud Playlists: 'int' object has no attribute 'startswith'

<img width="1920" height="897" alt="soundcloud error" src="https://github.com/user-attachments/assets/bc73022e-9932-4aee-9062-676df32a9988" />

**Reply by OzGav** [reply]
Please restart MA. Do the minimum required to get this to occur and then attach the full log here (MA SETTINGS>>SYSTEMS>>SERVER LOGGING )

**Reply by Traxtar928** [reply]
[music-assistant.log.txt](https://github.com/user-attachments/files/24501650/music-assistant.log.txt)

See attached log file.

Current version: 2.7.3
Installation method: Home Assistant OS
Core: 2026.1.0
Supervisor: 2026.01.0
Operating System: 16.3
Frontend: 20260107.0
Companion app: 2025.11.4-full (19134)

Steps to recreate: 

open MA
LIBRARY
BROWSE
SoundCloud
Playlists -> error

![Screenshot_20260108_121646_Home Assistant](https://github.com/user-attachments/assets/3de33b81-4a4b-413d-85cd-ac0a524bc85c)

**Reply by OzGav** [reply]
@robsonke

**Reply by robsonke** [reply]
Thanks for tagging. I can reproduce the issue too, I'll try to fix this asap.

**Reply by robsonke** [reply]
This should fix it: https://github.com/music-assistant/server/pull/2957

**Reply by Traxtar928** [reply]
Thanks all. 

Nothing for end users to do but wait for the next update to push through, correct?

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
